node/src/main/java/io/airlift/node/AddressToHostname.java

```diff
@@ -28,16 +30,39 @@ private AddressToHostname() {}
 
     public static String encodeAddressAsHostname(InetAddress inetAddress)
     {
-        String ipString = InetAddresses.toAddrString(inetAddress);
-        if (inetAddress instanceof Inet4Address) {
-            ipString = ipString.replace('.', '-');
-        }
-        else {
+        String hostname = switch (inetAddress) {
+            case Inet4Address inet4Address -> InetAddresses
+                    .toAddrString(inet4Address)
+                    .replace('.', '-');
+
             // v6 addresses contain `:` which isn't legal, also v6 can
             // start with a `:` so we prefix with an `x`
-            ipString = "x" + ipString.replace(':', '-');
+            case Inet6Address inet6Address -> 'x' + InetAddresses
+                    .toAddrString(inet6Address)
+                    .replace(':', '-')
+                    .replaceFirst(scopeWithDelimiter(inet6Address), "");
+
+            default -> InetAddresses.toAddrString(inetAddress);
+        };
+
+        return hostname + IP_ENCODED_SUFFIX;
+    }
+
+    // Same as in the https://github.com/google/guava/commit/3f61870ac6e5b18dbb74ce6f6cb2930ad8750a43
+    private static String scopeWithDelimiter(Inet6Address ip)
+    {
+        // getHostAddress on android sometimes maps the scope id to an invalid interface name; if the
```

Comment:
We don’t run on Android

Reply:
I wanted to keep this method the same as in Guava (since they are adding a scope id and we need to remove it)

Reply:
I've implemented it other way around (as suggested by the commit message in the Guava's change).